ignition-server/controllers/machineconfigserver_ignitionprovider.go

```diff
@@ -121,37 +135,45 @@ func (p *MCSIgnitionProvider) GetPayload(ctx context.Context, releaseImage strin
 		if mcsPod.Status.PodIP == "" || !mcsReady {
 			return false, nil
 		}
-
-		// Build proxy request.
-		proxyReq, err := http.NewRequest("GET", fmt.Sprintf("http://%s:%s/config/master", mcsPod.Status.PodIP, "8080"), nil)
-		if err != nil {
-			return false, fmt.Errorf("error building http request for machine config server pod: %w", err)
+		// // get the service
+		if err := p.Client.Get(ctx, ctrlclient.ObjectKeyFromObject(mcsService), mcsService); err != nil {
+			// We don't return the error here so we want to keep retrying.
+			return false, nil
 		}
-		// We pass expected Headers to return the right config version.
-		// https://www.iana.org/assignments/media-types/application/vnd.coreos.ignition+json
-		// https://github.com/coreos/ignition/blob/0cbe33fee45d012515479a88f0fe94ef58d5102b/internal/resource/url.go#L61-L64
-		// https://github.com/openshift/machine-config-operator/blob/9c6c2bfd7ed498bfbc296d530d1839bd6a177b0b/pkg/server/api.go#L269
-		proxyReq.Header.Add("Accept", "application/vnd.coreos.ignition+json;version=3.1.0, */*;q=0.1")
-
-		// Send proxy request.
-		client := &http.Client{
-			Timeout: 5 * time.Second,
+		//Generate Cert
+		if err := generateCert(mcsService); err != nil {
+			return false, fmt.Errorf("error generating cert for machine config server service: %w", err)
 		}
-		res, err := client.Do(proxyReq)
+
+		tlsConf := &tls.Config{}
```

Comment:
we should be using the basic http client and configuring the TLS parameter of it apropriately versus trying to read over the wire

Reply:
See this simple program for a example:
https://github.com/jcbsmpsn/golang-https-example/blob/master/https_client.go#L28

Reply:
CA cert to use in `rootCAs` is available at the `mcs-crt` secret. Specifically the `ca.crt` field of it